python_ta/checkers/global_variables_checker.py

```diff
@@ -76,6 +76,8 @@ def _get_child_disallowed_global_var_nodes(node):
     global, non-constant variable. 
     """
     node_list = []
+    if isinstance(node.parent, astroid.ListComp) or isinstance(node.parent, astroid.Comprehension):
```

Comment:
Two comments here:

1. If `ListComp` is necessary, then there should also be a check for `DictComp` and `SetComp` (there's a symmetry here).
2. But I think `node.parent` is the wrong approach (doesn't take into account nested expressions). I'd like to see you use `lookup_scope` (or whatever that method was called) to do this check.